src/api-binder/poll.ts

```diff
@@ -161,8 +171,9 @@ export const update = async (
 			body: body as any,
 		};
 
-		// Emit the target state and update the cache
-		cache.emitted = emitTargetState(cache, force, isFromApi);
+		// Emit the target state and update the cache, cancelling any
+		// apply operations that may be in progress
+		cache.emitted = emitTargetState(cache, force, isFromApi, true);
```

Comment:
I'm worried about a potential worsening of UX if the user sets a service var or config var during an image fetch. With this change as it stands right now, a variable change would equal a target state change, which would cause any current image fetches to be aborted.

Reply:
I think there is benefit of keeping the target state as a single entity as there may be interactions between config vars and updates (e.g. deltas). But I see your point and we can perhaps monitor any increase issues on support or other channels.